src/main/java/io/debezium/connector/spanner/kafka/internal/BufferedPublisher.java

```diff
@@ -69,16 +78,34 @@ public void buffer(V update) {
                 this.value.set(null);
                 this.onPublish.accept(update);
             }
+            if (update.getClass() == TaskSyncEvent.class) {
+                TaskSyncEvent event = (TaskSyncEvent) update;
+
+                LOGGER.info("Task {}, publishing event with rebalance generation ID {} and message type {}", this.taskUid,
+                        event.getRebalanceGenerationId(),
+                        event.getMessageType());
+            }
```

Comment:
Is there a specific reason why `BufferedPublisher` needs to do this if we only ever add `TaskSyncEvent` objects to the publisher?  I understand the tests use an `Integer`, but the test could be retrofitted to use a `TaskSyncEvent`, no?

Reply:
Sure, we can change it to be TaskSyncEvent specific. Since it requires a unittest refactoring, could we defer it as a future cleanup?

Reply:
Yes, please add it to the Jira proposed below